docs/dev/Pagination-v2.md

```diff
@@ -183,20 +573,75 @@ sequenceDiagram
-    QueryPlanFactory-->>-RestSQLQueryAction: UnsupportedCursorRequestException
+RestSQLQueryAction ->>+ SQLService : prepareRequest
+  SQLService ->>+ QueryPlanFactory : execute
+    rect rgb(191, 223, 255)
+    note over SQLService, CanPaginateVisitor : V2 support check
+    QueryPlanFactory ->>+ CanPaginateVisitor : canConvertToCursor
+      CanPaginateVisitor -->>- QueryPlanFactory : false
+    QueryPlanFactory -->>- RestSQLQueryAction : UnsupportedCursorRequestException
     deactivate SQLService
     end
-      RestSQLQueryAction->>Legacy Engine: accept
-      Note over Legacy Engine: Processing in Legacy engine
-        Legacy Engine-->>RestSQLQueryAction:complete
+      RestSQLQueryAction ->> Legacy Engine: accept
+      Note over Legacy Engine : Processing in Legacy engine
+        Legacy Engine -->> RestSQLQueryAction : complete
 ```
 
-### Serialization
+#### Serialization and Deserialization round trip
+
+The SQL engine should be able to completely recover the Physical Plan tree to continue its execution to get the next page. Serialization mechanism is responsible of proper dumping the plan tree. `ResourceMonitorPlan` shouldn't be serialized, because a new object of this type would be created for the restored plan tree before execution.
+Serialization and Deserialization are performed by Java object serialization API.
+
+## TODO describe ...
+
+```mermaid
+stateDiagram-v2
+    direction LR
+    state "Initial Query Request Plan Tree" as FirstPage
+    state FirstPage {
+        state "ProjectOperator" as logState1_1
+        state "..." as logState1_2
+        state "ResourceMonitorPlan" as logState1_3
+        state "OpenSearchPagedIndexScan" as logState1_4
+        state "OpenSearchScrollRequest" as logState1_5
+        logState1_1 --> logState1_2
+        logState1_2 --> logState1_3
+        logState1_3 --> logState1_4
+        logState1_4 --> logState1_5
+    }
+
+    state "Deserialized Plan Tree" as SecondPageTree
+    state SecondPageTree {
+        state "ProjectOperator" as logState2_1
+        state "..." as logState2_2
+        state "OpenSearchPagedIndexScan" as logState2_3
+        state "ContinuePageRequest" as logState2_4
+        logState2_1 --> logState2_2
+        logState2_2 --> logState2_3
+        logState2_3 --> logState2_4
+    }
+
+    state "Subsequent Query Request Plan Tree" as SecondPage
+    state SecondPage {
+        state "ProjectOperator" as logState3_1
+        state "..." as logState3_2
+        state "ResourceMonitorPlan" as logState3_3
+        state "OpenSearchPagedIndexScan" as logState3_4
+        state "ContinuePageRequest" as logState3_5
+        logState3_1 --> logState3_2
+        logState3_2 --> logState3_3
+        logState3_3 --> logState3_4
+        logState3_4 --> logState3_5
+    }
+
+  FirstPage --> SecondPageTree : Serialization and\nDeserialization
+  SecondPageTree --> SecondPage : Execution\nPreparation
+```
+
+#### Serialization
```

Comment:
Not mentioned here (although maybe we should): 
* What about the request context (like the query start time clock) - how is this serialized
* What about other request artefacts, such as `fetch_size` or maybe the `filter` - if provided.  How do these translate from request-to-request? 
* Does this work with post-request processing? 
* What about requests larger than the max window size of 10,000?

Reply:
- It is not supported yet, but actually it is just binary dumped into cursor
- It is not supported
- What do you mean?
- Good idea, I'll add

Reply:
Anything that we know is not-supported should be mentioned.